File: OrchidCore/src/main/java/com/eden/orchid/api/options/globals/ConfigGlobal.java
package com.eden.orchid.api.options.globals;

import com.eden.orchid.api.OrchidContext;
import com.eden.orchid.api.options.TemplateGlobal;

import java.util.Map;

public class ConfigGlobal implements TemplateGlobal<Map<String, Object>> {

    @Override
    public String key() {
        return "config";
    }

    @Override
    public Map<String, Object> get(OrchidContext context) {
        return context.getConfig();
    }
}

File: OrchidCore/src/main/kotlin/com/eden/orchid/api/options/globals/ConfigGlobal.kt
package com.eden.orchid.api.options.globals

import com.eden.orchid.api.OrchidContext
import com.eden.orchid.api.options.TemplateGlobal
import com.eden.orchid.api.theme.pages.OrchidPage

class ConfigGlobal : TemplateGlobal<Map<String?, Any?>?> {
    override fun key() = "config"
    override fun get(context: OrchidContext, page: OrchidPage?): Map<String?, Any?>? = context.config
}
